### ISSUE TITLE
It 2018 why are you using String?

### ISSUE THREAD

**Issue by mckeankylej** [harpocrates/language-rust#30]
[NO](https://github.com/harpocrates/language-rust/blob/98d208190e8473ffa472dce05216045764636819/src/Language/Rust/Data/Ident.hs#L73)
[It hurts](https://github.com/harpocrates/language-rust/blob/8900fc15978f52458439e215d6ea21e9f10f5fc4/src/Language/Rust/Parser/Literals.hs#L45)
*This being said thank you for using* `prettyprinter` *instead of* `ansi-wl-pprint`

**Comment by harpocrates** [comment]
The parser runs on `ByteString` by default, and the pretty printer will produce `Text` (or just stream to stdout) depending on the backend you prefer. What use of `String` are you objecting to in particular here?

If it is the one in `Ident`, I'd welcome a PR to fix it - provided the benchmarks show some improvement. Last time I tried naively swapping in `Text`, there was a _drop_ in performance. This is one situation where `String` is actually kind of what we want: the lexer produces one character at a time, so it isn't madness to just collect these into a list.

TL;DR Benchmarks please.

**Comment by mckeankylej** [comment]
I wasn't trying to be very serious with the issue so ima close it. 

On `Ident` even if the parser shows a decrease in performance it would be worth switching to `Text`. This is because thats a fixed cost for the user of this library. If `Ident` stays `String` then every time a library user uses ident they are wrought with cache misses and woe. 

Thanks for a great library!
